container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1877,6 +1884,36 @@ func (manager *KubernetesManager) waitForPodAvailability(ctx context.Context, na
 	)
 }
 
+// waitForPodDeletion waits for the pod to be fully deleted if it has been marked for deletion
+func (manager *KubernetesManager) waitForPodDeletion(ctx context.Context, namespaceName string, podName string) error {
+	// Wait for the pod to start running
+	deadline := time.Now().Add(podWaitForAvailabilityTimeout)
```

Comment:
if is necessary to reuse the constant I think that it could be renamed to support both cases